src/components/Authentication/LoginPage.js

```diff
@@ -125,6 +166,57 @@ const LoginPage = () => {
         }
     };
 
+    function refreshTokenSetup(res) {
+        return new Promise((resolve) => {
+            const refreshToken = async () => {
+                const newAuthRes = await res.reloadAuthResponse();
+                sessionStorage.setItem(
+                    'google_access_token',
+                    newAuthRes.access_token
+                );
+                resolve();
+            };
+            refreshToken();
+        });
+    }
+    const noGoogleCoursesFoundOnInitialGoogleLogin =
+        (google_courses === null || google_courses === undefined) &&
+        sessionStorage.getItem('google_access_token');
+    async function getCourses() {
+        if (noGoogleCoursesFoundOnInitialGoogleLogin) {
+            try {
+                const response = await axios.get(
+                    'https://classroom.googleapis.com/v1/courses',
+                    {
+                        headers: {
+                            Authorization: `Bearer ${sessionStorage.getItem(
+                                'google_access_token'
+                            )}`,
+                        },
+                    }
+                );
+                dispatch({
+                    type: actions.SET_GOOGLE_COURSES,
+                    payload: { google_courses: response?.data.courses },
+                });
+            } catch (error) {
+                console.log(error);
+            }
+        }
+    }
+
+    const onSuccess = (response) => {
+        setGoogleAuthEmail(response.profileObj.email);
+
+        getVerifyGoogleOauthTokenStatus();
+
+        refreshTokenSetup(response).then(() => {
+            getCourses();
+        });
+    };
+
+    const onFailure = () => {};
```

Comment:
we should do some actual error handling - either display a popup with the error message or just console.error it